binarySearch/binarySearch.js

```diff
@@ -10,6 +10,38 @@
  *	* console.log(index); // 1
 **/
 
+// binarySearch returns index of target number or -1 on failure
 const binarySearch = (nums, target) => {
 
+  const min = 0;
+  const max = nums.length;
+  const index = Math.floor((max - min) / 2);
+
+  // next index downward
+  const nextDown = (i) => {
+    if (i === min) return -1;
+    return Math.floor((i - min) / 2);
+  };
+
+  // next index upward
+  const nextUp = (i) => {
+    if (i === (max - 1)) return -1;
+    return i + Math.floor((max - i) / 2);
+  }
+
+  const bs = (i) => {
+    if (i === -1) return -1;
+    if (nums[i] === target) return i;
+    if (nums[i] > target) return bs(nextDown(i));
+    return bs(nextUp(i));
+  };
+
+  return bs(index);
```

Comment:
I love the recursive approach! This is a very common way to solve this problem. Your functions are nice and tidy and modular!